include/robot_localization/ros_filter.hpp

```diff
@@ -771,6 +771,16 @@ class RosFilter : public rclcpp::Node
   //!
   rclcpp::TimerBase::SharedPtr timer_;
 
+  //! @brief Time bookkeeping for periodic filter updates
+  //!
+  //! @details This is necessary to properly operate in simulation
+  //!
+  rclcpp::Time last_update_time_{0, 0, RCL_ROS_TIME};
```

Comment:
You're hardcoding a clock here? Shouldn't this be decided at runtime based on the node's clock?

Reply:
This is no longer necessary, but `robot_localization` ROS filter nodes always use `RCL_ROS_TIME` clocks (i.e. system time or simulation time depending on `use_sim_time` parameter value). It's true it's way better to query than to duplicate though.